bag.go

```diff
@@ -8,10 +8,93 @@ import (
 	"os"
 	"path/filepath"
 	"regexp"
+	"strings"
 )
 
-var manifestPtn = regexp.MustCompile("manifest-.*\\.txt$")
-var tagmanifestPtn = regexp.MustCompile("tagmanifest-.*\\.txt$")
+type Bag struct {
+	Path         string
+	Name         string
+	AbsPath      string
+	Payload      Payload
+	BagInfo      TagSet
+	BagIt        TagSet
+	Manifests    ManifestRefs
+	TagManifests ManifestRefs
+}
+
+func GetExistingBag(path string) (Bag, error) {
+	bag := Bag{}
+
+	//check directory exists
+	if err := directoryExists(path); err != nil {
+		return bag, err
+	}
+
+	//set path
+	bag.Path = path
```

Comment:
Would there be any issues if someone passed a relative path with `..`, e.g., `foo/../../bar/bag_root/`?

Reply:
I'll add a test